dask/array/core.py

```diff
@@ -2014,6 +2024,128 @@ def from_array(x, chunks, name=None, lock=False, asarray=True, fancy=True,
     return Array(dsk, name, chunks, dtype=x.dtype)
 
 
+def from_zarr(url, component=None, storage_options=None, chunks=None, **kwargs):
+    """Load array from the zarr storage format
+
+    See https://zarr.readthedocs.io for details about the format.
+
+    Parameters
+    ----------
+    url: str or MutableMapping
+        Location of the data. A URL can include a protocol specifier like s3://
+        for remote data. Can also be any MutableMapping instance, which should
+        be serializable if used in multiple processes.
+    component: str or None
+        If the location is a zarr group rather than an array, this is the
+        subcomponent that should be loaded, something like ``'foo/bar'``.
+    storage_options: dict
+        Any additional parameters for the storage backend (ignored for local
+        paths)
+    chunks: tuple of ints or tuples of ints
+        Passed to ``da.from_array``, allows setting the chunks on
+        initialisation, if the chunking scheme in the on-disc dataset is not
+        optimal for the calculations to follow.
+    kwargs: passed to ``zarr.Array``.
+    """
+    import zarr
+    storage_options = storage_options or {}
+    if isinstance(url, str):
+        fs, fs_token, path = get_fs_token_paths(
+            url, 'rb', storage_options=storage_options)
+        assert len(path) == 1
+        mapper = get_mapper(fs, path[0])
+    else:
+        mapper = url
+    z = zarr.Array(mapper, read_only=True, path=component, **kwargs)
+    chunks = chunks if chunks is not None else z.chunks
+    return from_array(z, chunks, name='zarr-%s' % url)
+
+
+def to_zarr(arr, url, component=None, storage_options=None,
```

Comment:
One more thing forgot to say. The ``url`` argument could default to ``None``. This would have the effect of creating a new in-memory array. User would have to also provide ``return_stored=True`` to be able to receive the new array, so there could be a gotcha with current default ``return_stored=False`` (maybe that should be True here?). 

Might be convenient in the case where you want to compute a result without having to bother about putting any data down on disk. I.e., given some Dask array ``d``, would be nice to be able to just do ``z = d.to_zarr()`` and get back an in-memory zarr array. In my work there's plenty of cases where I use in-memory zarr arrays, because it's quick and convenient and data are small enough with compression to fit in memory.

Don't mind if you'd rather leave as-is, just a thought.

Reply:
> The url argument could default to None. This would have the effect of creating a new in-memory array

I would think this is stretching the idea of what this method does, and be unexpected. A better approach maybe would be https://github.com/dask/dask/pull/2741 (i.e., a URL explicitly starting with `'memory://'`, but the reason that PR is languishing is that it's not clear what should happen in distributed memory.

Reply:
OK, no problem.

Reply:
Would just passing in a Zarr Array instance for `url` work for that case? If so, maybe this is already possible with a slightly different set of parameters. ;)

Reply:
FWIW I think it's good as-is. Might get a bit confusing if there is too much flexibility around what the url arg can be. And wouldn't gain much convenience, user would have to set up array themselves. My (lazy) use case is wanting to be able to do z = d.to_zarr(), which is like calling d.compute() but where the result is computed into an in-memory zarr array rather than numpy array. But happy to revisit that later, don't want to hold up this very nice PR :)

Reply:
FWIW I think it's good as-is. Might get a bit confusing if there is too much flexibility around what the url arg can be. And wouldn't gain much convenience, user would have to set up array themselves. My (lazy) use case is wanting to be able to do z = d.to_zarr(), which is like calling d.compute() but where the result is computed into an in-memory zarr array rather than numpy array. But happy to revisit that later, don't want to hold up this very nice PR :)